src/librustdoc/html/render.rs

```diff
@@ -1743,7 +1743,16 @@ fn item_module(w: &mut fmt::Formatter, cx: &Context,
             (Some(stability::Stable), Some(stability::Unstable)) => return Ordering::Less,
             _ => {}
         }
-        i1.name.cmp(&i2.name)
+        name_key(&i1.name).cmp(name_key(&i2.name))
+    }
+
+    fn name_key(name: &str) -> (&str, u64) {
+        let split = s.bytes().rposition(|b| b < b'0' || b'9' < b).unwrap_or(0);
```

Comment:
So we were actually both wrong on this one and I figured this out. The correct answer is `map_or(0, |s| s + 1)`.

Because if the position of the last non-digit is nothing, then this should be 0. Otherwise, it should be the position + 1 to be the beginning of the digits.